rules/AnnotationsToAttributes/Rector/Class_/RequiresAnnotationWithValueToAttributeRector.php

```diff
@@ -0,0 +1,245 @@
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @requires PHP > 8.4
+ * @requires PHPUnit >= 10
+ * @requires OS Windows
+ * @requires OSFAMILY Darwin
+ * @requires function someFunction
+ * @requires function \some\className::someMethod
+ * @requires extension mysqli
+ * @requires extension mysqli >= 8.3.0
+ * @requires setting date.timezone Europe/Berlin
+ */
+
+final class SomeTest extends TestCase
+{
+    /**
+     * @requires PHP > 8.4
+     * @requires PHPUnit >= 10
+     * @requires OS Windows
+     * @requires OSFAMILY Darwin
+     * @requires function someFunction
+     * @requires function \some\className::someMethod
+     * @requires extension mysqli
+     * @requires extension mysqli >= 8.3.0
+     * @requires setting date.timezone Europe/Berlin
+     */
+    public function test()
+    {
+    }
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+use PHPUnit\Framework\TestCase;
+
+#[\PHPUnit\Framework\Attributes\RequiresPhp('> 8.4')]
+#[\PHPUnit\Framework\Attributes\RequiresPhpunit('>= 10')]
+#[\PHPUnit\Framework\Attributes\RequiresOperatingSystem('Windows')]
+#[\PHPUnit\Framework\Attributes\RequiresOperatingSystemFamily('Darwin')]
+#[\PHPUnit\Framework\Attributes\RequiresFunction('someFunction')]
+#[\PHPUnit\Framework\Attributes\RequiresMethod(\some\className::class, 'someMethod')]
+#[\PHPUnit\Framework\Attributes\RequiresExtension('mysqli')]
+#[\PHPUnit\Framework\Attributes\RequiresExtension('mysqli', '>= 8.3.0')]
+#[\PHPUnit\Framework\Attributes\RequiresSetting('date.timezone', 'Europe/Berlin')]
+final class SomeTest extends TestCase
+{
+
+    #[\PHPUnit\Framework\Attributes\RequiresPhp('> 8.4')]
+    #[\PHPUnit\Framework\Attributes\RequiresPhpunit('>= 10')]
+    #[\PHPUnit\Framework\Attributes\RequiresOperatingSystem('Windows')]
+    #[\PHPUnit\Framework\Attributes\RequiresOperatingSystemFamily('Darwin')]
+    #[\PHPUnit\Framework\Attributes\RequiresFunction('someFunction')]
+    #[\PHPUnit\Framework\Attributes\RequiresMethod(\some\className::class, 'someMethod')]
+    #[\PHPUnit\Framework\Attributes\RequiresExtension('mysqli')]
+    #[\PHPUnit\Framework\Attributes\RequiresExtension('mysqli', '>= 8.3.0')]
+    #[\PHPUnit\Framework\Attributes\RequiresSetting('date.timezone', 'Europe/Berlin')]
+    public function test()
+    {
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class, ClassMethod::class];
+    }
+
+    public function provideMinPhpVersion(): int
+    {
+        return PhpVersionFeature::ATTRIBUTES;
+    }
+
+    /**
+     * @param Class_|ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->testsNodeAnalyzer->isInTestClass($node)) {
+            return null;
+        }
+
```

Comment:
Add `$hasChanged` flag here to be used to return later:

```suggestion
$hasChanged = false;
```